numba/cpython/unicode.py

```diff
@@ -2361,16 +2362,23 @@ def impl(n):
                 flag = True
             if n == 0:
                 return '0'
-            l = []
+            length = flag + 1 + int(np.floor(np.log10(n)))
+            kind = PY_UNICODE_1BYTE_KIND
+            char_width = _kind_to_byte_width(kind)
+            s = _malloc_string(kind, char_width, length, True)
+            const = ord('0')
+            if flag:
+                _set_code_point(s, 0, ord('-'))
+            idx = length - 1
             while n > 0:
-                c = chr(ord('0') + (n % ten))
```

Comment:
Is llvm not able to infer that `ord('0')` is a constant?

Reply:
expect it can, will just use it directly.